### PR TITLE
Add support for Sift Science's version-204 API.

### PR DESCRIPTION
**Pull request by jburnim**
This PR:

 - Adds support for calling multiple versions (203 vs 204) of the Sift Science API, where a user can specify which API to use at the client level or in individual method calls.

 - Adds support for calling the v204 API, with new parameters `abuse_type`, `abuse_types`, and `return_workflow_status`.

 - Adds support for the Workflow Status API, User Decisions API, and Order Decisions API.

 - Bumps the version of sift-python to 3.0.0.0, as the client now calls version 204 of the Sift API by default.

 - Tweaks the testing infrastructure.

@nickhurlburt -- could you please review these changes?

cc @atpaino , @fredsadaghiani , @yschatzberg

### CODE REVIEW

**Comment by jburnim** [comment]
Looks like test discovery was not added to the `unittest` module until Python 2.7.  I'll investigate switching to `unittest2` or only running the tests under 2.6 and 3.4.

File: sift/client.py [hunk 1]
```diff
@@ -77,35 +90,42 @@ def track(
                 event name such as "$transaction" or "$create_order" or a custom event
                 name (that does not start with a $).
 
-            properties: A dict of additional event-specific attributes to track
+            properties: A dict of additional event-specific attributes to track .
```
Comment (nickhurlburt) [inline sift/client.py]: Is the space before the period at the end of the line intentional?

File: sift/client.py [hunk 2]
```diff
@@ -77,35 +90,42 @@ def track(
...
             return_score: Whether the API response should include a score for this
-                 user (the score will be calculated using this event).  This feature must be
-                 enabled for your account in order to use it.  Please contact
-                 support@siftscience.com if you are interested in using this feature.
+                 user (the score will be calculated using this event).
 
             return_action: Whether the API response should include actions in the response. For
                  more information on how this works, please visit the tutorial at:
-                 https://siftscience.com/resources/tutorials/formulas
+                 https://siftscience.com/resources/tutorials/formulas .
```
Comment (nickhurlburt) [inline sift/client.py]: Also space-before-period here.

File: sift/client.py [hunk 3]
```diff
@@ -130,92 +153,121 @@ def track(
...
+    def label(self, user_id, properties, timeout=None, version=None):
         """Labels a user as either good or bad through the Sift Science API.
         This call is blocking.  Check out https://siftscience.com/resources/references/labels_api.html
         for more information on what fields to send in properties.
 
         Args:
             user_id:  A user's id. This id should be the same as the user_id used in
                 event calls.
+
             properties: A dict of additional event-specific attributes to track
```
Comment (nickhurlburt) [inline sift/client.py]: Do we need a period at the end of the line here?

**Comment by nickhurlburt** [comment]
LGTM.  What are your current thoughts on supporting Python 2.6 with respect to unit tests?

File: sift/client.py (hunk 1 above)
Comment (jburnim) [inline sift/client.py]: No -- I'll remove.

File: sift/client.py (hunk 2 above)
Comment (jburnim) [inline sift/client.py]: Yes -- this is to help ensure that browsers/editors/etc. that turn URLs into clickable links will point to [https://siftscience.com/resources/tutorials/formulas] and not [https://siftscience.com/resources/tutorials/formulas.].

File: sift/client.py (hunk 3 above)
Comment (jburnim) [inline sift/client.py]: I'll add one.

**Comment by jburnim** [comment]
> What are your current thoughts on supporting Python 2.6 with respect to unit tests?

I updated the Travis testing config to use unittest2.  Tests now pass for Python 2.6, 2.7, and 3.4: https://travis-ci.org/SiftScience/sift-python/builds/144594468 .

**Comment by nickhurlburt** [comment]
Awesome!  Looks ready to go, then.

**Comment by jburnim** [comment]
cc @aaronTufts

**Comment by jburnim** [comment]
@nickhurlburt , @aaronTufts -- I updated this PR to also add support for the `return_worfklow_status` Events API parameter, as well as the Workflow Status API, User Decisions API, and Order Decisions API.

**Comment by nickhurlburt** [comment]
Should workflow and decision methods have unit tests?

**Comment by jburnim** [comment]
> Should workflow and decision methods have unit tests?

Tests have been added.

**Comment by nickhurlburt** [comment]
Still LGTM, thanks!

**Pull request merged**